src/xb-builder.c

```diff
@@ -300,6 +300,10 @@ xb_builder_compile_source(XbBuilderCompileHelper *helper,
 
 	/* success */
 	xb_silo_add_profile(helper->silo, timer, "compile %s", guid);
+
+	/* free the istream, to release the file handle, because it's not needed anymore */
+	xb_builder_source_unset_istream(source);
```

Comment:
I do wonder if we could use g_ptr_array_copy() (with func==-NULL) and then remove it from the priv->sources GPtrArray instead...

Reply:
I'm not sure I follow. If you can call xb_builder_compile multiple times, then that would cause problems, no?

Reply:
Yes, but we can change the behaviour to disallow calling it multiple times with the opt-in XB_BUILDER_COMPILE_FLAG_INVALIDATE_SOURCES flag